autogpt_platform/backend/backend/util/exceptions.py

```
@@ -92,6 +94,43 @@ def __str__(self):
+    error: BaseException | str | None,
+    *,
+    allow_legacy_text: bool = False,
+) -> ExecutionFailureReason | None:
+    """Classify trusted exceptions, with an opt-in fallback for persisted text."""
+    if isinstance(error, InsufficientBalanceError):
+        return ExecutionFailureReason.INSUFFICIENT_BALANCE
+    if allow_legacy_text and isinstance(error, str):
+        if any(
+            pattern.fullmatch(error)
+            for pattern in _LEGACY_INSUFFICIENT_BALANCE_PATTERNS
+        ):
```

Ruff (0.16.1)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)